src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonParameterInfo.cs

```diff
@@ -101,14 +101,14 @@ public virtual void Initialize(JsonParameterInfoValues parameterInfo, JsonProper
                 // doesn't match the parameter type, use reflection to get the default value.
                 Type parameterType = parameterInfo.ParameterType;
 
-                GenericMethodHolder holder;
+                DefaultValueHolder holder;
```

Comment:
Ideally, we shouldn't even have this reflection-related code in code-paths that the source-gen feature flows through. The code for reflection/source-gen should be split/refactored from this [call site](https://github.com/dotnet/runtime/blob/65a5d0e8a3027f2b240d098f47327ba3e1f743b1/src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Metadata/JsonTypeInfo.Cache.cs#L649), but that can be done in a follow up.